packages/core/src/modules/message-pìckup/protocol/v2/V2MessagePickupProtocol.ts

```diff
@@ -82,21 +90,73 @@ export class V2MessagePickupProtocol extends BaseMessagePickupProtocol {
     return { message }
   }
 
+  public async deliverMessages(
+    agentContext: AgentContext,
+    options: DeliverMessagesProtocolOptions
+  ): Promise<DeliverMessagesProtocolReturnType<AgentMessage> | void> {
+    const { connectionRecord, recipientKey } = options
+    connectionRecord.assertReady()
+
+    const messagePickupRepository = agentContext.dependencyManager.resolve<MessagePickupRepository>(
+      InjectionSymbols.MessagePickupRepository
+    )
+
+    // Get available messages from queue, but don't delete them
+    const messages = await messagePickupRepository.takeFromQueue({
+      connectionId: connectionRecord.id,
+      recipientKey: recipientKey,
+      limit: 10, // TODO: Define as config parameter
+      keepMessages: true,
+    })
```

Comment:
Should we somehow add an option to mark them as delivered? but not remove them from the queue yet (so they won't be re-delivered directly, but they also aren't removed). Maybe something for future updates